docs/paper/dev/api/command-api/basics/command-registration.mdx

```diff
@@ -0,0 +1,133 @@
+the `registrar` one. With it we can register our commands.
+
+
+### Registering inside a plugin main class
+Getting access to a `LifecycleEventManager` for commands inside our plugin's main class is very similar to how you access it inside the PluginBootstrap class, with the difference
+that instead of getting the `LifecycleEventManager` using the `BootstrapContext` provided to us in the bootstrap method of our PluginBootstrap class, we can just retrieve it using
+`JavaPlugin#getLifecycleManager`.
+
+```java title="PluginMainClass.java"
+public final class PluginMainClass extends JavaPlugin {
+
+    @Override
+    public void onEnable() {
+        this.getLifecycleManager().registerEventHandler(LifecycleEvents.COMMANDS, commands -> {
+            // register your commands here ...
+        });
+    }
+}
+```
+
+This follows the same concept as the PluginBootstrap one, just that instead of being given a `LifecycleEventManager<BootstrapContext>`, we are instead given a 
+`LifecycleEventManager<Plugin>`. This doesn't really matter for our use cases, but you might as well be aware of that.
+The rest of the methods works the exact same way as with the `PluginBootstrap` parameterized `LifecycleEventManager`.
+
+## Registering Commands using the Commands class
+Now that we have access to the instance of a `Commands` class via `commands.registrar()` in our event handler, we have access to a few overloads of the `Commands#register`
+method.
+
+### Registering a LiteralCommandNode
+Most of the time, you will be using a `LiteralArgumentBuilder` to build up your command tree. In order to retrieve a `LiteralCommandNode` from that object, we need to call the
+`LiteralArgumentBuilder#build()` method on it:
+
+```java
+LiteralArgumentBuilder<CommandSourceStack> command = Commands.literal("testcmd")
+    .then(Commands.literal("argument_one"))
+    .then(Commands.literal("argument_two"));
+
+LiteralCommandNode<CommandSourceStack> buildCommand = command.build();
+```
+
+Or in short:
+
+```java
+LiteralCommandNode<CommandSourceStack> buildCommand = Commands.literal("testcmd")
+    .then(Commands.literal("argument_one"))
+    .then(Commands.literal("argument_two"))
+    .build();
+```
+
+Now that we have retrieved our `LiteralCommandNode`, we can register it. For that we have multiple overloads, which optionally allow us to set aliases and/or the description.
+Registering our "testcmd" might look like this:
+
+```java
+this.getLifecycleManager().registerEventHandler(LifecycleEvents.COMMANDS, commands -> {
+    commands.registrar().register(buildCommand);
+});
+```
+
+### Registering a BasicCommand
+A <Javadoc name='io.papermc.paper.command.brigadier.BasicCommand'>`BasicCommand`</Javadoc> is a bukkit-like way of defining commands. Instead of building up a command tree, we allow all user input and retrieve the arguments as a simple array of Strings.

```

Comment:
```suggestion
A <Javadoc name='io.papermc.paper.command.brigadier.BasicCommand'>`BasicCommand`</Javadoc> is a Bukkit-like way of defining commands. Instead of building up a command tree, we allow all user input and retrieve the arguments as a simple array of strings.
```